controllers/componentversion_controller.go

```diff
@@ -75,107 +75,111 @@ func (r *ComponentVersionReconciler) Reconcile(ctx context.Context, req ctrl.Req
-			conditions.Set(component, reconciling)
+			conditions.Set(obj, reconciling)
 		}
 
 		// If not reconciling or stalled than mark Ready=True
-		if !conditions.IsReconciling(component) &&
-			!conditions.IsStalled(component) &&
+		if !conditions.IsReconciling(obj) &&
+			!conditions.IsStalled(obj) &&
 			retErr == nil &&
-			result.RequeueAfter == component.GetRequeueAfter() {
-			conditions.MarkTrue(component, meta.ReadyCondition, meta.SucceededReason, "Reconciliation success")
+			result.RequeueAfter == obj.GetRequeueAfter() {
+			conditions.MarkTrue(obj, meta.ReadyCondition, meta.SucceededReason, "Reconciliation success")
 		}
 		// Set status observed generation option if the component is stalled or ready.
-		if conditions.IsStalled(component) || conditions.IsReady(component) {
-			component.Status.ObservedGeneration = component.Generation
+		if conditions.IsStalled(obj) || conditions.IsReady(obj) {
+			obj.Status.ObservedGeneration = obj.Generation
 		}
 
 		// Update the object.
-		if err := patchHelper.Patch(ctx, component); err != nil {
+		if err := patchHelper.Patch(ctx, obj); err != nil {
 			retErr = errors.Join(retErr, err)
 		}
 	}()
 
 	//TODO@souleb: reduce logging verbosity, use events instead. This will be easier to use logs
 	// for debugging and events for monitoring
-	log.V(4).Info("found component", "component", component)
+	log.V(4).Info("found component", "component", obj)
 
 	// reconcile the version before calling reconcile func
-	update, version, err := r.checkVersion(ctx, component)
+	update, version, err := r.checkVersion(ctx, obj)
 	if err != nil {
 		retErr = fmt.Errorf("failed to check version: %w", err)
 		return *result, retErr
 	}
 
 	if !update {
-		log.V(4).Info("no new version which satisfies the semver constraint detected")
 		result = &ctrl.Result{
-			RequeueAfter: component.GetRequeueAfter(),
+			RequeueAfter: obj.GetRequeueAfter(),
 		}
 		return *result, nil
 	}
 
 	log.Info("running verification of component")
-	ok, err := r.OCMClient.VerifyComponent(ctx, component, version)
+	ok, err := r.OCMClient.VerifyComponent(ctx, obj, version)
 	if err != nil {
 		err := fmt.Errorf("failed to verify component: %w", err)
-		conditions.MarkStalled(component, v1alpha1.VerificationFailedReason, err.Error())
-		conditions.MarkFalse(component, meta.ReadyCondition, v1alpha1.VerificationFailedReason, err.Error())
+		conditions.MarkStalled(obj, v1alpha1.VerificationFailedReason, err.Error())
+		conditions.MarkFalse(obj, meta.ReadyCondition, v1alpha1.VerificationFailedReason, err.Error())
 		retErr = err
 		return *result, retErr
 	}
 
 	if !ok {
 		err := fmt.Errorf("attempted to verify component, but the digest didn't match")
-		conditions.MarkStalled(component, v1alpha1.VerificationFailedReason, err.Error())
-		conditions.MarkFalse(component, meta.ReadyCondition, v1alpha1.VerificationFailedReason, err.Error())
+		conditions.MarkStalled(obj, v1alpha1.VerificationFailedReason, err.Error())
```

Comment:
When we are marking as Stalled should we be returning the error? Stalled indicates that the controller will not requeue the object because it has reached a stage where user intervention is required to fix the problem. But if we return an error it will requeue.

Reply:
Hmm, good point.